### ISSUE TITLE
getPointerProps Returns Previous Selected Point Index on Release

### ISSUE THREAD

**Issue by simyxing** [Abhinandan-Kushwaha/react-native-gifted-charts#854]
### Description

When using the `getPointerProps` function, the index updates correctly as the pointer moves across different points on the chart. However, upon releasing the pointer, the index reverts to the previously selected point instead of returning index -1. 

### Expected Behavior
Upon releasing the pointer, getPointerProps should return index = -1 to signify no point is selected after release.

### Actual Behavior:
On release, the index reverts to the previously selected point, rather than returning -1.

### Workaround:
Temporarily, I am just using `pointerX = 0` to identify if it is released.

### Additional Information:
![image](https://github.com/user-attachments/assets/0f73e068-760d-4ca0-84b8-a677d3c409e8)

### Steps to reproduce

1. Install library
2. Import <LineGraph /> component
3. add props
```
  getPointerProps={(item: {pointerIndex: number; pointerX: number;}) => {
    updatePoint (item.pointerX === 0 ? -1 : item.pointerIndex)
  }};
```

### version of react-native-gifted-charts

1.4.34

### React Native version

0.75.2

### Platforms

Android, iOS

### Workflow

React Native

**Comment by Abhinandan-Kushwaha** [comment]
Hi @simyxing Can you check again in the latest version of react-native-gifted-charts?

**Comment by simyxing** [comment]
@Abhinandan-Kushwaha Yes, the issue is persist in the latest version of react-native-gifted-charts (1.4.43)

**Comment by jeanniton-mnr** [comment]
I am using the latest version of react native gifted charts.

I have found out upon releasing the pointer on a bar chart, `BarChart`, the pointer index correctly returns  `-1`.

However, on a line chart, `LineChart`, upon releasing the pointer the same last point index  is erroneuosly is returned.

I will look at the source code. If am able to fix that I will submit a fix merge request.

react-native-gifted-charts is one of the best chart library I have found for react-native.

Thanks!

**Comment by Abhinandan-Kushwaha** [comment]
Hi @jeanniton-mnr Thanks for suggesting the fix. Your PR is merged and I have added a new property named `resetPointerIndexOnRelease`. Now you just need to add `resetPointerIndexOnRelease: true` in the `pointerConfig` object to get -1 as pointerIndex upon release.

This fix is available from versions `1.4.48` onwards. Please use the latest version of the library.
